src/integrationTest/java/com/mongodb/hibernate/query/select/temporal/AbstractSelectionTemporalIntegrationTest.java

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright 2025-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.query.select.temporal;
+
+import static java.lang.String.format;
+
+import com.mongodb.hibernate.query.AbstractQueryIntegrationTests;
+import java.util.List;
+import java.util.Set;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+abstract class AbstractSelectionTemporalIntegrationTest<I, T> extends AbstractQueryIntegrationTests {
+
+    static final String COLLECTION_NAME = "items";
+
+    @BeforeEach
+    void beforeEach() {
+        getSessionFactoryScope().inTransaction(session -> getData().forEach(session::persist));
+        getTestCommandListener().clear();
+    }
+
+    abstract List<I> getData();
+
+    @ParameterizedTest(name = "testComparisonByEq: temporal={0}, expectedItems={2}, expectedRender={3}, timeZone={1}")
```

Comment:
There is no `timeZone` parameter here, or in any other test methods declared in this class, which suggests that something is wrong with the file.

I skipped reviewing `AbstractSelectionTemporalIntegrationTest` and `SelectionInstantIntegrationTest`. I am asking to please at first self-review them, taking into account the rest of the comments in this PR, and do the changes, if needed.

`TODO` for @stIncMale: review what was skipped.